SudokuProject/src/cspSolver/BTSolver.java

```diff
@@ -277,6 +279,59 @@ private Variable getMRV()
 	 */
 	private Variable getDegree()
 	{
+		List<Variable> tempNetwork = network.getVariables();
+		int maxDegree = 0;
+		Variable tempVar = tempNetwork.get(0);
+		for(Variable v: tempNetwork){
+			if(!v.isAssigned()){
+				int tempDegree = getDegree(v);
+				if(tempDegree > maxDegree){
+					maxDegree = tempDegree;
+					tempVar = v;
+				}
+			}
+		}
+		if(maxDegree != 0){
+			return tempVar;
+		}
+		return null;
+	}
+	/**
+	 *  This function gets the degree of a variable
+	 */
+	private int getDegree(Variable v){
+		int degree = 0;
+		for(Variable w: network.getNeighborsOfVariable(v)){
+			if(!w.isAssigned()){
+				degree++;
+			}
+		}
+		return degree;
+	}

```

Comment:
This is purely just code DRYness but there's probably a method we can call on the array returned by ```getNeighborsOfVariable(v)``` to count how many objects have a certain condition false.

EDIT: Damn just looked it up and it's not worth it. The simple ```.filter``` method is new to Java 8 so we can't use it.